From 185bc588997eb15fbd0be67766379f3f0cedea8a Mon Sep 17 00:00:00 2001
From: GloriousEggroll <gloriouseggroll@gmail.com>
Date: Tue, 13 May 2025 15:41:09 -0600
Subject: [PATCH 1/2] proton: move hdr settings to their own envvar for those
 that want wayland but dont have or need hdr

Use PROTON_ENABLE_HDR to use HDR with Wayland

fixes: CachyOS/proton-cachyos#21
---
 proton | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/proton b/proton
index 52e65b89..b047f04f 100755
--- a/proton
+++ b/proton
@@ -1592,6 +1592,7 @@ class Session:
         self.check_environment("PROTON_FORCE_LARGE_ADDRESS_AWARE", "forcelgadd")
         self.check_environment("PROTON_OLD_GL_STRING", "oldglstr")
         self.check_environment("PROTON_ENABLE_WAYLAND", "enablewayland")
+        self.check_environment("PROTON_ENABLE_HDR", "enablehdr")
         self.check_environment("PROTON_HIDE_NVIDIA_GPU", "hidenvgpu")
         self.check_environment("PROTON_HIDE_VANGOGH_GPU", "hidevggpu")
         self.check_environment("PROTON_HIDE_INTEL_GPU", "hideintelgpu")
@@ -1627,8 +1628,9 @@ class Session:
             if self.env.get("WAYLAND_DISPLAY", False):
                 self.dlloverrides["winex11.drv"] = ""
                 self.dlloverrides["winewayland.drv"] = "b"
-                self.env["DXVK_HDR"] = "1"
-                self.env["ENABLE_HDR_WSI"] = "1"
+                if "enablehdr" in self.compat_config:
+                    self.env["DXVK_HDR"] = "1"
+                    self.env["ENABLE_HDR_WSI"] = "1"
                 self.env.pop("DISPLAY", "")
 
         if os.environ.get("SteamGameId", 0) in [
-- 
2.49.0

